awx/main/dispatch/pool.py

```diff
@@ -365,6 +367,28 @@ def cleanup(self):
                 logger.warn('scaling down worker pid:{}'.format(w.pid))
                 w.quit()
                 self.workers.remove(w)
+            if w.alive:
+                # if we discover a task manager invocation that's been running
+                # too long, reap it (because otherwise it'll just hold the postgres
+                # advisory lock forever); the goal of this code is to discover
+                # deadlocks or other serious issues in the task manager that cause
+                # the task manager to never do more work
+                current_task = w.current_task
+                if current_task:
+                    if current_task.get('task', '').endswith('tasks.run_task_manager'):
```

Comment:
I get confused in the dispatcher code about whether these messages are always dicts, or sometimes strings.

Well if `current_task` is a string, I don't really care what it does. Just don't do anything would be fine.